### PR TITLE
Ils fix report

### PR DESCRIPTION
**Pull request by lwyszomi**

### CODE REVIEW

File: custom/ilsgateway/tanzania/reports/delivery.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 from django.db.models.aggregates import Avg
 from corehq.apps.locations.models import SQLLocation
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
-from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter
+from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter, ProgramFilter
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/reports/delivery.py]: F401 'ProductByProgramFilter' imported but unused

File: custom/ilsgateway/tanzania/reports/facility_details.py [hunk 2]
```diff
@@ -135,27 +136,69 @@ def rows(self):
...
     def headers(self):
         return DataTablesHeader(*[
-            DataTablesColumn()
+            DataTablesColumn('Date'),
+            DataTablesColumn('Direction'),
+            DataTablesColumn('Text')
         ])
 
 
+    @property
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/reports/facility_details.py]: E303 too many blank lines (2)

File: custom/ilsgateway/tanzania/reports/randr.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 from corehq.apps.locations.models import SQLLocation, Location
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
 from corehq.apps.users.models import CommCareUser
-from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter
+from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter, ProgramFilter
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/reports/randr.py]: F401 'ProductByProgramFilter' imported but unused

File: custom/ilsgateway/tanzania/reports/supervision.py [hunk 4]
```diff
@@ -3,12 +3,12 @@
 from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
 from corehq.apps.reports.filters.fixtures import AsyncLocationFilter
 from corehq.apps.reports.filters.select import YearFilter
-from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter
+from custom.ilsgateway.filters import ProductByProgramFilter, MSDZoneFilter, MonthAndQuarterFilter, ProgramFilter
```
Comment (dimagimon) [inline custom/ilsgateway/tanzania/reports/supervision.py]: F401 'ProductByProgramFilter' imported but unused

**Comment by dimagimon** [comment]
There are 1 errors:

* custom/ilsgateway/tanzania/__init__.py, line 230 - E303 too many blank lines (2)

**Comment by lwyszomi** [comment]
@czue

**Comment by czue** [comment]
didn't review super closely but looks good overall. fancy use of celery apis!

File: custom/ilsgateway/tanzania/reports/facility_details.py [hunk 5]
```diff
@@ -136,27 +137,68 @@ def rows(self):
...
         return DataTablesHeader(*[
-            DataTablesColumn()
+            DataTablesColumn('Date'),
+            DataTablesColumn('Direction'),
+            DataTablesColumn('Text')
         ])
 
+    @property
+    def rows(self):
+        data = SMSLog.by_domain_date(self.config['domain'])
```
Comment (snopoke) [inline custom/ilsgateway/tanzania/reports/facility_details.py]: @lwyszomi @kkrampa this should use the `corehq.apps.sms.models.SMS` SQL model and do the location filtering in SQL

Comment (snopoke) [inline custom/ilsgateway/tanzania/reports/facility_details.py]: and the sorting

**Pull request merged**